### PR TITLE
Ported env test to e2e

### LINKED ISSUE #3354: [E2E] port integration test "environment" to the e2e framework
## E2E Task

Port:
 - https://github.com/sylabs/singularity/blob/master/cmd/singularity/env_test.go
to the e2e framework
 - https://github.com/sylabs/singularity/tree/master/e2e
under `e2e/actions`

### PR DESCRIPTION
**Pull request by schebro**
**Description of the Pull Request (PR):**

Ported env test to E2E

**This fixes or addresses the following GitHub issues:**

- Fixes #3354 

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

**Review by jscook2345 — APPROVED** [review]
LGTM

File: e2e/env/env.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+type testingEnv struct {
+	// base env for running tests
+	CmdPath string `split_words:"true"`
+}
+
+var testenv testingEnv
+
+func singularityEnv(t *testing.T) {
+	// Singularity defines a path by default. See singularityware/singularity/etc/init.
```
Comment (WestleyK) [inline e2e/env/env.go:30]: >// Singularity defines a path by default. See singularityware/singularity/etc/init.

Is this necessary? or should the url be updated to https://github.com/sylabs/singularity/blob/master/etc/init?

File: e2e/env/env.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	for _, currentTest := range singularityEnvTests {
+		t.Run(currentTest.name, test.WithoutPrivilege(func(t *testing.T) {
+			args := []string{"exec", currentTest.image, "env"}
+
+			cmd := exec.Command(testenv.CmdPath, args...)
+			cmd.Env = append(os.Environ(), currentTest.env...)
+			b, err := cmd.CombinedOutput()
+
+			out := string(b)
+			t.Logf("args: '%v'", strings.Join(args, " "))
```
Comment (WestleyK) [inline e2e/env/env.go:69]: This for sure is a string, so you can use:

```suggestion
			t.Logf("args: '%s'", strings.Join(args, " "))
```

File: e2e/env/env.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+			t.Logf("env: '%v'", strings.Join(cmd.Env, " "))
```
Comment (WestleyK) [inline e2e/env/env.go:70]: Here to:

```suggestion
			t.Logf("env: '%s'", strings.Join(cmd.Env, " "))
```

**Review by WestleyK — COMMENTED** [review]
Couple suggestions...

**Review by ikaneshiro — APPROVED** [review]
lgtm once these minor changes are addressed

File: e2e/env/env.go (hunk 1 above)
Comment (jscook2345) [reply e2e/env/env.go:30]: @WestleyK Yeah, that should be updated. I put that in there before our repository migration I believe.

**Pull request closed without merging**